src/Nethermind/Nethermind.Core/Collections/CollectionExtensions.cs

```diff
@@ -40,6 +40,12 @@ public static bool NoResizeClear<TKey, TValue>(this ConcurrentDictionary<TKey, T
 
             using var handle = dictionary.AcquireLock();
 
+            // Recheck under lock
+            if (dictionary is null || dictionary.IsEmpty)
```

Comment:
`dictionary?.IsEmpty == false`

Reply:
> dictionary?.IsEmpty == false

Crashes with that 

`dictionary?.IsEmpty ?? true` works though